platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/roster/RosterValidatorTests.java

```diff
@@ -0,0 +1,305 @@
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build(),
+                                RosterEntry.newBuilder()
+                                        .nodeId(3)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .build())
+                        .build()));
+    }
+
+    @Test
+    void zeroPortTest() {
+        assertThrows(
+                InvalidRosterException.class,
+                () -> RosterValidator.validate(Roster.newBuilder()
+                        .rosters(
+                                RosterEntry.newBuilder()
+                                        .nodeId(1)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(0)
+                                                .build())
+                                        .build(),
+                                RosterEntry.newBuilder()
+                                        .nodeId(2)
+                                        .weight(1)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build(),
+                                RosterEntry.newBuilder()
+                                        .nodeId(3)
+                                        .weight(2)
+                                        .gossipCaCertificate(Bytes.wrap("test"))
+                                        .tssEncryptionKey(Bytes.wrap("test"))
+                                        .gossipEndpoint(ServiceEndpoint.newBuilder()
+                                                .domainName("domain.com")
+                                                .port(666)
+                                                .build())
+                                        .build())
+                        .build()));
+    }
+
+    @Test
+    void domainAndIpTest() {
+        assertThrows(
```

Comment:
i think the ipv6 question applies here too. IF we support it, we should have a test case for it.

Reply:
Nope, please see above. However, I'm adding a check for the length of the ip address field.